python/pecos/simulators/quantum_simulator.py

```diff
@@ -61,6 +61,8 @@ def run(self, qops: list[QOp]) -> list:
         """
         meas = []
         for op in qops:
+            if op.metadata is None:
+                op.metadata = {}
```

Comment:
Alternatively, perhaps, the default value of `op.metadata` could be `{}` in the constructor.

Reply:
I am not quite sure which constructor you mean. I think this probably serves both the `QuantumCircuit` and PHIR-based data structures. I would say, unless it is causing an issue, for now leave it as is as there probably needs to be addressed in another wave of refactoring. I would like to simplify and get rid of this but it might break other things.